sdk/storage/Microsoft.Azure.WebJobs.Extensions.Storage.Blobs/src/Listeners/BlobTargetScalerProvider.cs

```diff
@@ -0,0 +1,114 @@
+            BlobServiceClient blobServiceClient = blobServiceClientProvider.Get(blobMetadata.Connection, serviceProvider.GetRequiredService<INameResolver>());
+            _targetScaler = new ZeroToOneTargetScaler(triggerMetadata.FunctionName, blobServiceClient, blobMetadata.Path, factory);
+        }
+
+        public ITargetScaler GetTargetScaler()
+        {
+            return _targetScaler;
+        }
+
+        internal class ZeroToOneTargetScaler : ITargetScaler
+        {
+            private readonly TargetScalerDescriptor _targetScalerDescriptor;
+            private readonly Lazy<Task<BlobLogListener>> _blobLogListener;
+            private readonly ILogger _logger;
+            private readonly string _containerName;
+            private StorageAnalyticsLogEntry _latestPositiveEntry = null;
+
+            public ZeroToOneTargetScaler(string functionId, BlobServiceClient blobServiceClient, string blobPath, ILoggerFactory loggerFactory)
+            {
+                _targetScalerDescriptor = new TargetScalerDescriptor(functionId);
+                _blobLogListener = new(() => BlobLogListener.CreateAsync(
+                    blobServiceClient,
+                    loggerFactory.CreateLogger<BlobListener>(),
+                    CancellationToken.None));
+                _logger = loggerFactory.CreateLogger<ZeroToOneTargetScaler>();
+                int separatorIndex = blobPath.IndexOf("/", StringComparison.OrdinalIgnoreCase);
+                if (separatorIndex > 0)
+                {
+                    _containerName = blobPath.Substring(0, separatorIndex);
+                }
+                else
+                {
+                    throw new ArgumentException($"The blob path '{blobPath}' is not valid. It should be in format '<container-name>/<path-to-blob>'.");
+                }
+            }
+
+            public TargetScalerDescriptor TargetScalerDescriptor => _targetScalerDescriptor;
+
+            public async Task<TargetScalerResult> GetScaleResultAsync(TargetScalerContext context)
+            {
+                var now = DateTimeOffset.UtcNow;
+
+                var startOfPreviousHour = new DateTime(now.Year, now.Month, now.Day, now.Hour, 0, 0).AddHours(-1);
+                if (_latestPositiveEntry?.RequestStartTime > startOfPreviousHour)
+                {
+                    _logger.LogInformation($"Recent writes were detected from cache for '{_targetScalerDescriptor.FunctionId}' (requestedObjectKey='{_latestPositiveEntry.RequestedObjectKey}', requestStartTime='{_latestPositiveEntry.RequestStartTime:O}').");
+                    return new TargetScalerResult() { TargetWorkerCount = 1 };
+                }
+
+                var blobLogListener = await _blobLogListener.Value.ConfigureAwait(false);
+                _latestPositiveEntry = await blobLogListener.GetFirstWriteForContainerAsync(_containerName, CancellationToken.None).ConfigureAwait(false);
```

Comment:
Above, looks like you're only considering writes within the last hour. But on this call you're not passing in a value for the hoursWindow param so it looks back 2 hours unnecessarily?